CRM/Contribute/BAO/Contribution.php

```diff
@@ -193,8 +193,7 @@ public static function add(&$params, $ids = array()) {
       $params['prevContribution'] = self::getOriginalContribution($contributionID);
     }
 
-    // CRM-16189
-    CRM_Financial_BAO_FinancialAccount::checkFinancialTypeHasDeferred($params, $contributionID);
+    CRM_Core_BAO_FinancialTrxn::generateRevenueRecognitionDate($params, $contributionID);
```

Comment:
@pradpnayak I encountered a fatal error while doing backoffice contribution. 
```
PHP Fatal error:  Call to undefined method CRM_Core_BAO_FinancialTrxn::generateRevenueRecognitionDate() in /Users/monish/src/civicrm/CRM/Contribute/BAO/Contribution.php on line 196
```
didn't find any definition of it in core or is there any other dependent PR for this?